v0/destinations/hs/util.js

```diff
@@ -300,6 +302,151 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
+      update => update.property === destinationExternalId
+    );
+
+    if (filteredInfo.length) {
+      input.message.context.externalId = setHsSearchId(
+        input,
+        filteredInfo[0].id
+      );
+      input.message.context.hubspotOperation = "update";
+      resultInput.push(input);
+    } else {
+      input.message.context.hubspotOperation = "create";
+      resultInput.push(input);
+    }
+  });
+  return resultInput;
+};
+
+const getHsSearchId = message => {
+  let externalIdArray = [];
+  let hsSearchId = null;
+  externalIdArray = message.context?.externalId;
+
+  if (externalIdArray) {
+    externalIdArray.forEach(extIdObj => {
+      const { type } = extIdObj;
+      if (type.includes("HS")) {
+        hsSearchId = extIdObj.hsSearchId;
+      }
+    });
+  }
+  return { hsSearchId };
+};
+
+const setHsSearchId = (input, id) => {
+  const { message } = input;
+  const resultExternalId = [];
+  if (message.context && message.context.externalId) {
+    externalIdArray = message.context.externalId;
+  }
+  if (externalIdArray) {
+    externalIdArray.forEach(extIdObj => {
+      const { type } = extIdObj;
+      if (type.includes("HS")) {
+        extIdObj.hsSearchId = id;
+      }
+      resultExternalId.push(extIdObj);
+    });
+  }
+  return resultExternalId;
+};
+
+const getExistingData = async (inputs, destination) =>{
+  const { Config } = destination;
+  const values = [];
+  let searchResponse;
+  const updateHubspotIds = [];
+  const firstMessage = inputs[0].message;
+  let objectType = null;
+  let identifierType = null;
+
+  if (firstMessage) {
+    objectType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .objectType;
+    identifierType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .identifierType;
+  }
+  inputs.map(async input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+    values.push(destinationExternalId);
+  });
+  let requestData = {
+    filterGroups: [
+      {
+        filters: [
+          {
+            propertyName: identifierType,
+            values,
+            operator: "IN"
+          }
+        ]
+      }
+    ],
+    properties: [identifierType],
+    limit: 100,
+    after: 0
+  };
+
+  const requestOptions = {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${Config.accessToken}`
+    }
+  };
+  let checkAfter = 1; // variable to keep checking if we have more results
+  while (checkAfter) {
+    const endpoint = IDENTIFY_CRM_SEARCH_ALL_OBJECTS.replace(
+      ":objectType",
+      objectType
+    );
+    const url =
+      Config.authorizationType === "newPrivateAppApi"
+        ? endpoint
+        : `${endpoint}?hapikey=${Config.apiKey}`;
+    searchResponse =
+      Config.authorizationType === "newPrivateAppApi"
+        ? await httpPOST(url, requestData, requestOptions)
```

Comment:
clarification: error scenarios to be handled or It will be thrown to caller?

Reply:
you mean errors from http post request?